contracts/Question.sol

```diff
@@ -49,20 +50,28 @@ contract Question {
     donorCount = 1;
     donations = amount;
     donorAmounts[author] = amount;
+    allDonors.push(author);
     highestDonors[0] = HighestDonor({ addr: author, lastDonatedAt: now });
   }
+  
+  function getDonorAmountByKey (uint key) public constant returns(donation uint) {
+    return donorAmounts[allDonors[key]];
+  }
 
   function receiveApproval(address from, uint256 value, address _tokenContract, bytes extraData) beforeDeadline {
-    require(address(token) == _tokenContract);
+    require(address(token) == _tokenContract); // necessary?
     require(value > 0);
-    require(token.transferFrom(from, this, value));
+    require(token.transferFrom(from, this, value)); // why is from being used instead of msg.sender?
```

Comment:
As far as we using `approveAndCall` method of Aeternity token contract, `msg.sender` will be equal to address of the token contract, instead of the address of the user who is making a donation.